### DISCUSSION TITLE
about dash-cam

### DISCUSSION THREAD

**Discussion by jh-martin-1** [gopro/labs#1033] [Q&A] [answered]
请问，这两个USB Power Trigger代码有什么不同？谢谢！
!MBOOT="!Lbt"!SAVEbt=<u0!X"Dashcam"=At:B=C0>u0<r0!C8+!S>u0=At:B+=C0<u0>r0=Bt:B+=CB+=C-A+"wait $Cs">C9>r0!E+!1N+!1O<r0"Exit Dashcam"+!X!R34，这个来自于：https://github.com/gopro/labs/discussions/1006
*FAST=1*WAKE=2*BOOT="!Lbt"!SAVEbt=<u0!X<r0!S>u0=At:B<u0>r0=Bt:B+=B-A>B10>r0!E+!1N+!1O<r0!X!R10，这个来自于：https://gopro.github.io/labs/control/usb/

**Comment by dnewman-gpsw** [comment]
Second one doesn't auto clear media, otherwise variations on the same idea.

**Comment by jh-martin-1** [comment]
请问，black10可以做到边充电边录像吗？

**Reply by dnewman-gpsw** [reply]
No, that I understand is a hardware limitation.

**Comment by jh-martin-1** [comment]
刚搞明白，Visual Script Tool真是太棒了，基本上可以搞定我的需求。谢谢老师！

**Comment by jh-martin-1** [comment]
您好！
!MBOOT="!Lbt"!SAVEbt=<u0!X<r0"Charge..."+!10N+!S>u0=At:B<u0>r0=Bt:B+=B-A>B9>r0!E+!1N+!1O<r0!X!R27
当sleep x seconds中的x大于9时（比如10），相机出现提示，并会不断地关闭并重新启动，只有x小于9才能正常启动并录像，这是什么原因呢？谢谢！
![20240823100406](https://github.com/user-attachments/assets/5d99910d-f295-41a7-b1f5-ebd919584a85)

**Comment by dnewman-gpsw** [comment]
Correct, that is by design.  Longer sleeps, it is more effecicent to shutdown.  !9N or less will not sleep.  You can do !9N!9N

**Comment by jh-martin-1** [comment]
非常感谢。谢谢您这么快就回复我！

**Comment by jh-martin-1** [comment]
您好！我想在相机开始录像前响两声，设置如图，但是，它却没有录像变成了拍照，这是什么原因呢？不设置!B2，它是可以立即录像的。谢谢！
!MBOOT="!Lbt"!SAVEbt=<u0!X<r0oV9+!B2+!S+oV0>u0=At:B<u0>r0=Bt:B+=B-A>B9>r0!E+!1N+!1O<r0!X!R22

![beep](https://github.com/user-attachments/assets/ec887154-992f-446c-b83d-f4b78b5a3ba2)

**Reply by dnewman-gpsw** [reply]
It will take photos if your camera defaults to photo. In the menus set default video, or add 'mV' to the script to set video mode.

**Comment by jh-martin-1** [comment]
![xjys](https://github.com/user-attachments/assets/bdd487c3-3c1f-4cd2-b0f5-4c7a5bbe4b89)
![xjys1](https://github.com/user-attachments/assets/fc9714d0-4bc4-40cd-9d6a-da992e954a8e)
您好！试过了，我的相机black10手动开机进入的是默认模式（视频），但是QR Code加了!B2后插入USB自动开机就进入拍照模式，不加!B2，就正常开始视频录制了，不知什么原因？
QR Code：!MBOOT="!Lbt"!SAVEbt=<u0!X<r0!B2+!S>u0=At:B<u0>r0=Bt:B+=B-A>B9>r0!E+!1N+!1O<r0!X!R14

**Comment by jh-martin-1** [comment]
您看一下视频，就是这个样子。我是想要相机响一声，它却拍照了，为什么？

https://github.com/user-attachments/assets/09e283f4-eba4-4231-8263-f39e9a9fb0dc

**Reply by dnewman-gpsw** [reply, marked as answer]
Ah! You have found a Labs bug in HERO10, this will work on H11 and H12, but H10 the !B command is still enabling the burst mode.

**Reply by jh-martin-1** [reply]
是否可以修正它？

**Reply by dnewman-gpsw** [reply]
Yes, but it is not likely. The newer model get more frequent updates, so get more Labs releases. HERO10 is soon to be 3 generations old, with no currently scheduled releases planned.

**Reply by jh-martin-1** [reply]
产品还在售，又是刚买的，希望您能抽点时间修复它。

**Reply by dnewman-gpsw** [reply]
Labs is only for prototypes, it is beta by design, so nothing is guaranteed to be a polished or working bonus feature.

**Reply by jh-martin-1** [reply]
理解！
请问，是否可以尝试把HERO11的Labs固件应用在10上？
谢谢！

**Reply by dnewman-gpsw** [reply]
No, all camera have different firmware, the hardware is different, e.g. HERO11 has twice the memory of HERO10.

**Reply by jh-martin-1** [reply]
好的。如果使用了会是什么后果？GoPro的Labs，是我选购GoPro相机的重要因素，它不同于其它的运动相机，它能让我们diy自己的想法。所以，我还是要说，既然我们已经发现了bug，那就应该修复它，否则，我们如何乐享其中呢！

**Reply by dnewman-gpsw** [reply]
It will not install. Enjoy as is.

**Reply by jh-martin-1** [reply]
享受bug，No！新的Labs固件何时更新？谢谢您！

**Reply by dnewman-gpsw** [reply]
There is a low probability to be updated, particularly of a cosmetic bug (technically not a bug even, !B is for burst mode on older cameras.)  The newest model will get the next Labs update, then maybe another for HERO12, rare possibility for HERO11. HERO10 just diminishingly too old to get any more scheduled updates. HERO10 has had 9 Labs releases to date, that was a good run. Labs uses the release cycles of the main firmware, it almost never releases on it own, as each release cost many thousands of dollars in engineering and QA effort. If you see an update on HERO10 main firmware, there will be a Labs update. If having the latest Labs features is really important, consider upgrading your camera, lots of cool stuff is coming.

**Reply by jh-martin-1** [reply]
!B不是必须要的功能，HERO10Black当作一个行车记录仪使用最重要的工作是：让它能够持续录制10小时或者更长时间（已经做到了），因此不会考虑升级它,。谢谢您！
